src/cpu_feature/cpu_feature.rs

```diff
@@ -0,0 +1,61 @@
+extern crate libc;
+
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+use std::fs::File;
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+use std::io::BufReader;
+// TODO represent unsigned long without libc
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+use self::libc::c_ulong;
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+use self::auxv::{AuxValError, AuxVals};
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+use self::cpuinfo::CpuInfo;
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+use self::arm::arm_cpuid_setup;
+
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+static HAS_BROKEN_NEON: bool = false;
+
+// TODO is this used in practice? Its C equivalent is the only user of the
+// global, and the function name has no uses
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+#[allow(non_snake_case)]
+extern "C" fn GFp_has_broken_NEON() -> bool {
+    return HAS_BROKEN_NEON;
+}
+
+extern "C" {
+    #[cfg(all(target_os = "linux", target_arch = "arm"))]
+    #[allow(non_upper_case_globals)]
+    pub static mut GFp_armcap_P: u32;
+}
+
+#[cfg(all(target_os = "linux", target_arch = "arm"))]
+#[allow(non_snake_case)]
+extern "C" fn GFp_cpuid_setup() {
+    if let Ok(r) = File::open("/proc/cpuinfo").map(|f| BufReader::new(f)) {
```

Comment:
See the suggestion below. All of this code is cross-platform except the reference to /proc/cpuinfo, so we should move it all to the cross-platform cpuinfo subcrate, possibly having `parse_cpuinfo` open the file and read it itself.